### DISCUSSION TITLE
device/subdevice numbers

### DISCUSSION THREAD

**Discussion by mark3smith** [probonopd/irdb#77] [Q&A] [answered]
Could anyone give a quick explanation of device/subdevice numbers?  Is there a human readable mapping somewhere?  Thx.

**Comment by bengtmartensson** [comment, marked as answer]
An IR protocol is a method of generating IR signals given the values of its parameters. There are often, but not always, called _function_ (or "F"), _device_  (or "D"), and _subdevice_ (or "S"). You can generate ("render") a protocol for particular values of the parameters using for example [IrScrutinizer](https://github.com/bengtmartensson/IrScrutinizer)

![Screenshot from 2021-09-08 18-25-03](https://user-images.githubusercontent.com/11508922/132548056-0a3053a1-9dec-45e7-b68b-8a1f6dfcd016.png)

The inverse operation (given an IR signal, determine a protocol and associated parameter values that correspond to the given signal (if any)) is called _decoding_, and can also be done by IrScrutinizer.

**Reply by mark3smith** [reply]
Makes sense.  How would would typically know which file to use?  Would he have to use IrScrutinizer (or something) to capture a code from an existing remote to read the function/subfunction?  Or is there a map of IR remote model numbers that maps to the function/subfunction?

**Reply by bengtmartensson** [reply]
That is the way IRDB was conceived... Sometimes you even need several file for a particular device.  (See #58).

You can of course just try, until you find something that works.  Please also try RemoteLocator (#63).

**Comment by probonopd** [comment]
As far as I know this notation originates in DecodeIR `protocols.ini`, http://www.hifi-remote.com/forums/viewtopic.php?t=14219.
